src/Room.js

```diff
@@ -212,7 +213,16 @@ Room.prototype.renew = function (filters, notificationCB, cb) {
   self.kuzzle.subscriptions.pending[self.id] = self;
 
   subscribeQuery.body = self.filters;
-  subscribeQuery = self.kuzzle.addHeaders(subscribeQuery, this.headers);
+  subscribeQuery = self.kuzzle.addHeaders(subscribeQuery, self.headers);
+
+  networkListener = function() {
```

Comment:
I don't like this behaviour (too many "addListener" and "removeListener").

Better just add following lines into `src/Kuzzle.js` (at line [365](https://github.com/kuzzleio/sdk-javascript/blob/321211f442a14dad8136a9ed4d3dee112905d63f/src/Kuzzle.js#L365)) :
```javascript
self.network.onConnectError(function (error) {
// (...)
  disableAllSubscriptions.call(self);
// (...)
});

function disableAllSubscriptions() {
  var self = this;

  Object.keys(self.subscriptions).forEach(function (roomId) {
    Object.keys(self.subscriptions[roomId]).forEach(function (subscriptionId) {
      var subscription = self.subscriptions[roomId][subscriptionId];
      subscription.subscribing = false;
    });
  });
}
```

It should work, but without a complex listener management.